transcom/mymove
# INT- v2 - B-21608 alaska update search counseling office query

## [B-21608](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1027048)
## [INT PR](https://github.com/transcom/mymove/pull/14172)

## Summary

This PR is a follow-up to [INT PR](https://github.com/transcom/mymove/pull/14172) relating to issues uncovered during PO acceptance. It resolves the following:

1.  Excluding JPPSO offices from the counseling offices dropdown
2. Updated SQL to compute distance to properly sort offices from closest to furthest. NOTE: UI will still not display correct sort order because the data to compute distances are not present in table: zip3_distances. Beth Grohmann is aware and currently is looking into migrating this data when data is available.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. From the customer app, create users with following affiliations: AIR FORCE, MARINES
2. Login to onboard a move. Search for "Fort Wainwright, AK" duty location using users in step 1. Verify counseling office dropdown does not display any JPPSO offices. See list below for list of offices per GBLOC.
3. [Reference original INT PR](https://github.com/transcom/mymove/pull/14172)  for list of valid duty locations.

NOTE: sorting issue will still be present. Sql has been updated but the data needed for correct sorting still has not been migrated. That work is still to be determined.

**GBLOC Office Mappings**:
**MBFL**:
PPPO Eielson AFB
PPPO JBER Travel Center- Elmendorf
PPPO JBER Travel Center- Richardson

**JEAT**:
PPPO Fort Leavenworth - USA
PPPO Fort Sill - USA
PPPO McChord Field - USA
PPPO Fort Greely
PPPO Fort Wainwright
PPPO Fort Leonard Wood - USA
PPPO NAVSUP FLC PS Everett - USN
PPPO NAVSUP FLC PS Whidbey Island - USN
PPPO JB Lewis-McChord (Fort Lewis) - USA
PPPO NAVSUP FLC Puget Sound - USN
PPPO Base Seattle - USCG
PPPO Fort Riley - USA

**MAPS**:
PPPO BSU Kodiak

**MAPK**:
PPPO Base Ketchikan


### Backend
